Makefile

```diff
@@ -0,0 +1,18 @@
+.PHONY: docker_stop docker_start check_docker check_docker_compose run_test test
+
+check_docker: ; @which docker > /dev/null
+
+check_docker_compose: ; @which docker-compose > /dev/null
+
+docker_start:
+	@cd docker; docker-compose up -d;
+
+docker_stop:
+	@cd docker; docker-compose down
+
+run_test:
+	@cd docker; docker-compose exec go-diameter bash -c  "cd go-diameter/ && go test ./..."
+
+test: check_docker check_docker_compose docker_start run_test
+	$(info )
+	$(info --- run `make docker_stop` to stop the test container --- )
```

Comment:
This makefile only deals with docker based tests, let's move it under docker directory.

go-diameter itself neither requires docker, nor make to build & test. We shouldn't be adding additional requirements.

Reply:
The tests doesn't pass in mac. So I guess having docker as a standard platform to make sure the code doesn't have issue is better than just not passing tests